test/models/pregnancy_test.rb

```diff
@@ -43,4 +43,35 @@ class PregnancyTest < ActiveSupport::TestCase
       assert @pregnancy.created_by
     end
   end
+  
+  before do
+    @pregnancy.dcaf_soft_pledge = 500
+    @pt_1.clinic_name = 'Nice Clinic'
+    @pt_1.appointment_date = DateTime.now + 14    
+  end
+
+  describe 'pledge_sent validation' do
+    it 'should not validate pledge_sent if the DCAF soft pledge field is blank' do
+      @pregnancy.dcaf_soft_pledge = nil
+      @pregnancy.pledge_sent = true
+      refute @pregnancy.valid?
```

Comment:
Oh man this is great, A++++ work. Love how readable and clear this is. 

Can you confirm in these sets of tests that the right error message comes back too? I think that'll keep the test coverage at 100%, and we've got some modal stuff coming up that will consume that.